tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py

```diff
@@ -186,20 +186,38 @@ def run(self):
                         )
                         return False
                     time.sleep(10)
-                    finished = True
+                    child_job_finished_count = 0
+                    child_job_failed = []
                     for child_job_id in parent_job.children:
                         child_job = NS.tendrl.objects.Job(
                             job_id=child_job_id
                         ).load()
-                        if child_job.status != "finished":
-                            finished = False
-                            break
-                    if finished:
+                        if child_job.status == "finished":
+                            child_job_finished_count += 1
+                        if child_job.status == "failed":
+                            child_job_finished_count += 1
+                            child_job_failed.append(child_job.job_id)
+                    if child_job_finished_count == len(node_list) - 1:
```

Comment:
earlier case if anyone child job fails then it is marking parent job as failed but the parent is checking only child finished or not it is not checking child is failed, so parent job is waiting until timeout.

So the user may be thought a job is failed and let's go with unmanage, but unmanage flow will fail because one storage node is still stuck with import job. 

here i am checking finished node + failed node count if it is equal to a number of child node then i am stopping job.